packages/react-components/react-swatch-picker-preview/docs/Spec.md

```diff
@@ -2,62 +2,395 @@
 
 ## Background
 
-_Description and use cases of this component_
+A SwatchPicker is used in graphic and text editors.
+It allows user to choose a needed color, image or pattern.
+
+The SwatchPicker can be integrated within a popover or used as a standalone feature.
 
 ## Prior Art
 
-_Include background research done for this component_
+- [Convergence epic](https://github.com/microsoft/fluentui/issues/28606)
+
+### Fabric (v8)
+
+```jsx
+import { IColorCellProps, SwatchColorPicker } from '@fluentui/react/lib/SwatchColorPicker';
+
+const colorCellsExample = [
+  { id: 'a', label: 'orange', color: '#ca5010' },
+  { id: 'b', label: 'cyan', color: '#038387' },
+  { id: 'c', label: 'blueMagenta', color: '#8764b8' },
+  { id: 'd', label: 'magenta', color: '#881798' },
+  { id: 'e', label: 'white', color: '#ffffff' },
+];
+
+export const SwatchColorPickerBasicExample: React.FunctionComponent = () => {
+  const [previewColor, setPreviewColor] = React.useState<string>();
+  const baseId = useId('colorpicker');
+
+  const swatchColorPickerOnCellHovered = (id: string, color: string) => {
+    setPreviewColor(color!);
+  };
 
-- _Link to Open UI research_
-- _Link to comparison of v7 and v0_
-- _Link to GitHub epic issue for the converged component_
+  return (
+    <>
+      <div id={`${baseId}-custom-size`}>Simple square swatch color picker</div>
+      <SwatchColorPicker
+        columnCount={5}
+        cellHeight={35}
+        cellWidth={35}
+        cellShape={'square'}
+        colorCells={colorCellsExample}
+        onCellHovered={swatchColorPickerOnCellHovered}
+        aria-labelledby={`${baseId}-custom-size`}
+      />
+    </>
+  );
+};
+```
+
+### 3rd party Design Systems
+
+- Adobe Spectrum
+  - [SwatchGroup](https://spectrum.adobe.com/page/swatch-group/)
+  - [Swatch](https://spectrum.adobe.com/page/swatch/)
+
+### Components
+
+| Purpose                                                      | Fabric (V8)         | V9           | Matching? |
+| ------------------------------------------------------------ | ------------------- | ------------ | --------- |
+| Component responsible for rendering swatches as row and grid | SwatchColorPicker   | SwatchPicker | ⚠️        |
+| Color cell                                                   | ColorPickerGridCell | ColorSwatch  | ⚠️        |
+| Image cell                                                   |                     | ImageSwatch  | ❌        |
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```jsx
+<SwatchPicker aria-label="Font color" layout="grid" columnCount={3}>
```

Comment:
wondering if we should introduce an explicit `label` prop that would set aria-label under the hood, to explicitly force users to define accessible label ?

Reply:
I'll think about it. I didn't want to introduce additional props because `aria-label` is known prop and that what users expect to be there. Also, if they don't provide it then `value` will be used instead

Reply:
> Also, if they don't provide it then value will be used instead

which value ? from child component ? 

> I didn't want to introduce additional props because aria-label is known prop

fair enough. will aria-label trigger native tooltip hint when hovered/focused over ? if not then I think we should encapsulate this under `label`  (at least for child components)

Reply:
I was referring to title attribute but it looks like it's not very good idea to use it https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title#accessibility_concerns.